test/parallel/test-vm-inherited_properties.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+require('../common');
+
+const vm = require('vm');
+const assert = require('assert');
+
+let base = {
+  propBase: 1
+};
+
+let sandbox = Object.create(base, {
+  propSandbox: { value: 3 }
+});
+
+const context = vm.createContext(sandbox);
+
+let result = vm.runInContext('Object.hasOwnProperty(this, "propBase");',
+                             context);
+
+assert.strictEqual(result, false);
+
+// Ref: https://github.com/nodejs/node/issues/5350
+base = Object.create(null);
+base.x = 1;
+base.y = 2;
+
+sandbox = Object.create(base);
+sandbox.z = 3;
+
+assert.deepStrictEqual(Object.keys(sandbox), ['z']);
+
+const code = 'x = 0; z = 4;';
+result = vm.runInNewContext(code, sandbox);
+assert.strictEqual(result, 4);
+
+assert.deepStrictEqual(Object.keys(sandbox), ['z', 'x']);
+
+// Check that y is not an own property.
+assert.strictEqual((Object.keys(sandbox)).indexOf('y'), -1);
```

Comment:
I think we can remove this assert. It's implicitly checked on line 37.

Reply:
You're right. Deleted. 

Now that most of the vm issues are fixed, we could go through and consolidate all the tests.